### ISSUE TITLE
Remove (or at least deprecate) complex number methods

### ISSUE THREAD

**Issue by AlexDBlack** [deeplearning4j/nd4j#2624]
Have have (and have had, for a while) some complex number methods in ND4J.
Such as a lot of Nd4j.createComplex() methods, Nd4j.complexX() methods.

Given that these aren't maintained and don't really (at all?) work, we should at a minimum deprecate them all, or even better remove them before the next release

**Comment by raver119** [comment]
Yes, probably. We're going to bring their support together with other datatypes, but that'll be part of SameDiff api i think. So all existing stuff will be significantly changed.
